tests/nightly/test_all.sh

```diff
@@ -72,9 +80,15 @@ check_val() {
 example_dir=../../example/image-classification
 # python: lenet + mnist
 test_lenet() {
-    python $example_dir/train_mnist.py \
-        --data-dir `pwd`/data/mnist/ --network lenet --gpus $gpus --num-epochs 10 \
-        2>&1 | tee log
+    optimizer=$(get_random_optimizer)
```

Comment:
This test does not deterministically find issues.  Can we run instead on all optimizers ? may be a separate test if needed